bowling/src/tokenizer.cpp

```diff
@@ -0,0 +1,23 @@
+#include <algorithm>
+#include "tokenizer.hpp"
+
+const char Tokenizer::FrameDelimiterSign = '|';
+const char Tokenizer::NameDelimiterSign = ':';
+
+std::string Tokenizer::readSequence(const std::string& line, size_t& position) {
+    std::string result = {std::next(line.begin() + position), line.end()};
+
+    result.erase(std::remove(result.begin(), result.end(), FrameDelimiterSign), result.end());
+
+    return result;
+}
+
+std::string Tokenizer::readName(const std::string& line, size_t& position) {
+    size_t next = line.find(NameDelimiterSign);
+    if (next != std::string::npos) {
+        position = next;
+        return line.substr(0, next);
+    } else {
+        return "";
+    }
+}
```

Comment:
It took me a while to understand how these functions work. 

I don't know if it's a good idea to make it like this. Now `readSequence()` and `readName()` methods are tightly coupled by the `position` parameters and the way how they are called in FrameParser class. Wouldn't be better to just remove `position` from a parameters and once again use `line.find(NameDelimiterSign)` ? I know that it's less efficient but readability and **independence** (no coupling between) of these methods will be better.

So it would be something like this:
```cpp
std::string Tokenizer::readSequence(const std::string& line) {
    size_t position = line.find(NameDelimiterSign);
    if (position == std::string::npos) {
        return "";
    }
    std::string result = {std::next(line.begin() + position), line.end()};
    result.erase(std::remove(result.begin(), result.end(), FrameDelimiterSign), result.end());
    return result;
}

std::string Tokenizer::readName(const std::string& line) {
    size_t position = line.find(NameDelimiterSign);
    if (position != std::string::npos) {
        return line.substr(0, position);
    } else {
        return "";
    }
}
```

And then in `FrameParser::parse`  the `size_t currentLinePosition{};` is not needed.

And of course tokenizer-ut will change
```cpp
TEST_P(GetNameSuite, TestReadNameReturnsPartOfTheStringToColonSign) {
    auto [param, expected] = GetParam();
    ASSERT_EQ(Tokenizer::readName(param), expected);
}

TEST_P(GetSeqSuite, TestReadSequenceReturnsStringWithoutNameAndDelimiter) {
    auto [param, expected] = GetParam();
    ASSERT_EQ(Tokenizer::readSequence(param), expected);
}
```

Or maybe we can find the `NameDelimiterSign` position once and store it somewhere for future use? But finding this delimiter should **not** be done in `readName` method for later use in `readSequence`. Maybe additional `findNameDelimiter()` method should be created?

Reply:
Done.